src/Session/SessionManager.cc

```diff
@@ -139,406 +141,28 @@ void SessionManager::OnMessage(WSType* ws, std::string_view sv_message, uWS::OpC
-                    break;
-                }
-                case CARTA::EventType::SET_VECTOR_OVERLAY_PARAMETERS: {
-                    CARTA::SetVectorOverlayParameters message;
-                    if (message.ParseFromArray(event_buf, event_length)) {
-                        tsk = new GeneralMessageTask<CARTA::SetVectorOverlayParameters>(session, message, head.request_id);
-                        message_parsed = true;
-                    }
-                    break;
-                }
-                case CARTA::EventType::STOP_FITTING: {
-                    CARTA::StopFitting message;
-                    if (message.ParseFromArray(event_buf, event_length)) {
-                        session->OnStopFitting(message);
-                        message_parsed = true;
-                    }
-                    break;
-                }
-                case CARTA::EventType::STOP_PV_PREVIEW: {
-                    CARTA::StopPvPreview message;
-                    if (message.ParseFromArray(event_buf, event_length)) {
-                        session->OnStopPvPreview(message);
-                        message_parsed = true;
-                    }
-                    break;
-                }
-                case CARTA::EventType::CLOSE_PV_PREVIEW: {
-                    CARTA::ClosePvPreview message;
-                    if (message.ParseFromArray(event_buf, event_length)) {
-                        session->OnClosePvPreview(message);
-                        message_parsed = true;
-                    }
-                    break;
-                }
-                case CARTA::EventType::REMOTE_FILE_REQUEST: {
-                    CARTA::RemoteFileRequest message;
-                    if (message.ParseFromArray(event_buf, event_length)) {
-                        session->OnRemoteFileRequest(message, head.request_id);
-                        message_parsed = true;
-                    }
-                    break;
-                }
-                case CARTA::EventType::CHANNEL_MAP_FLOW_CONTROL: {
-                    CARTA::ChannelMapFlowControl message;
-                    if (message.ParseFromArray(event_buf, event_length)) {
-                        session->HandleChannelMapFlowControlEvt(message);
-                        message_parsed = true;
-                    }
-                    break;
-                }
-                default: {
-                    spdlog::warn("Bad event type {}!", event_type);
-                    break;
-                }
-            }
-
-            if (!message_parsed) {
-                spdlog::warn("Bad {} message!", event_type_name);
-            }
-
-            if (tsk) {
-                ThreadManager::QueueTask(tsk);
+        if (handler != _message_handlers.end()) {
+            auto start_time = std::chrono::high_resolution_clock::now();
+            try {
+                handler->second(session, event_buf, event_length, head);
+            } catch (const std::exception& e) {
```

Comment:
This should only be catching the message parsing exception, for consistency with the old code (other exceptions should not be handled here; there are generic exception handlers at a higher level).